src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,370 @@
+                .cpus(cpusPerNode)
+                .groups(userGroups)
+                .build();
+    }
+
+    private List<Queue> fillQueueNameFromOutput(final List<String> resultOutput) {
+        return resultOutput.stream()
+                .map(partitionData -> {
+                    final String[] partitionDataArray = partitionData.split(STANDARD_SLURM_DELIMETER);
+                    if (partitionDataArray.length != SCONTROL_OUPUT_SIZE) {
+                        throw new GridEngineException(HttpStatus.NOT_FOUND, "Partition output data is incorrect");
+                    }
+                    return partitionDataArray[SCONTROL_OUPUT_PARTNAME_INDEX].trim();
+                })
+                .distinct()
+                .map(partitionName -> Queue.builder()
+                        .name(partitionName)
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    private List<Queue> fillQueueData(final List<String> resultOutput) {
+        final List<SlurmQueue> partitionsData = resultOutput.stream()
+                .map(this::getPartitionData)
+                .collect(Collectors.toList());
+
+        return partitionsData.stream().map(
+                        partitionData -> Queue.builder()
+                                .name(partitionData.getPartition())
+                                .hostList(partitionData.getNodelist())
+                                .allowedUserGroups(partitionData.getGroups())
+                                .slots(new SlotsDescription(
+                                        partitionData.getCpus() * partitionData.getNodelist().size(),
+                                        getNodesDescription(partitionData.getNodelist(), partitionData.getCpus())
+                                ))
+                                .build())
+                .collect(Collectors.toList());
+    }
+
+    private Map<String, Integer> getNodesDescription(final List<String> nodes, final int cpus) {
+        if (nodes.isEmpty()) {
+            return new HashMap<>();
+        }
+        return nodes.stream()
+                .collect(Collectors.toMap(
+                        nodeListElem -> nodeListElem,
+                        nodeListElem -> cpus)
+                );
+    }
+
+    private Context prepareContext(final QueueVO registrationRequest) {
+        final Context context = new Context();
+        context.setVariable(SCONTROL_ACTION, SCONTROL_CREATE_COMMAND);
+        if (registrationRequest.getAllowedUserGroups() != null) {
+            context.setVariable(SCONTROL_USER_GROUPS, registrationRequest.getAllowedUserGroups());
+        }
+        if (registrationRequest.getHostList() != null) {
+            context.setVariable(SCONTROL_HOSTS, registrationRequest.getHostList());
+        }
+        context.setVariable(SCONTROL_PARTITION_NAME, registrationRequest.getName());
+        return context;
+    }
+
+    private List<String> expandHosts(final String hosts) {
```

Comment:
better to name like `mapHostRangeToHosts`

Reply:
and since we already use regexp, why not use it completely?

So:
1) in line [215](https://github.com/SilinPavel/grid-engine-api/pull/18/files#diff-61a7dd70c8c5e1258c3bded95ff7c17a66ace18d10403a2ecc0a2832d0042f38R215) just call `mapHostsStringToHosts(host)`
2) here, in `mapHostsStringToHosts` check if string matches regexp
3) If not match just return list with single item
4) if matches - get all information from matcher:
   - change regexp to `"([a-zA-Z]+)\\[(\\d+)-(\\d+)\\]"`
   - after calling a `matcher.find()` you can safely call `matcher.group(<number-of-group>)` to get node name and ranges, read more about it here: https://medium.com/stackera/java-regex-part-6-group-and-subgroup-2985dc2d42d4

Reply:
done